File: golden-square-solo-project/lib/order.rb
class Order
  attr_reader :submitted

  def initialize(customer, cart, time = Time)
    @customer = customer
    @cart = cart
    @date_time = 0
    @submitted = false
    @time = time
  end

  def submit
    fail if @submitted
    # sends a text to the customer
    @date_time = @time.now
    @submitted = true  # sends order to kitchen
  end

  def customer
    @customer
  end

  def cart
    @cart
  end

  def date_time
    fail 'Order not submitted!' if @date_time == 0
    @date_time.strftime("%d/%m/%Y, %k:%M")
  end

  def get_order_time
    @date_time
  end
end

File: java-takeaway-challenge/app/src/main/java/OrderSystem/Order.java
package OrderSystem;

import java.time.Clock;
import java.time.LocalDateTime;

public class Order {
  private Cart cart;
  private Customer customer;
  private LocalDateTime orderTime;

  public Order(Cart cart, Customer customer) {
    this.cart = cart;
    this.customer = customer;
    orderTime = null;
  }

  public LocalDateTime getOrderTime() {
    return orderTime;
  }

  public void submit(Clock clock) {
    orderTime = LocalDateTime.now(clock);
  }

  public Customer getCustomer() {
    return customer;
  }

  public Cart getCart() {
    return cart;
  }
}
